src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -176,6 +190,15 @@ void DirectoryService::ComputeSharding(
     for (const auto& kv : sortedPoWSolns)
```

Comment:
I think this might have a performance issue once we hit, say, 10k nodes.
It might be better to have two loops: one for if `setTopPriorityNodes` is empty, and an else loop for otherwise.
In the if branch, you have a for loop over `sortedPoWSolns`.
In the else branch, you have a for loop over `setTopPriorityNodes`.

Reply:
At most of the time, the FindTopPriorityNodes function will not be called, so setTopPriorityNodes will be empty, and the checking setTopPriorityNodes is empty is very fast.